### ISSUE TITLE
Diff of rebased PatchSet is not correctly shown

### ISSUE THREAD

**Issue by jbaumgar** [uwolfer/gerrit-intellij-plugin#290]
When a PatchSet is rebased on another change, the diff does not recognise the rebase and also shows changes which are not caused by the PatchSet but by the rebase.

For example:
PatchSet 42 of a change is a rebase (e.g rebased by Rebase Button in Gerrit Web UI), No changes are made in PatchSet 42. When PatchSet 42 is seletect in Clion/Gerrit Plugin and "Diff against" is set to the previous patch set 41, the diff shows the diffs of a file which is changed in the commit the patch set is rebased on.
When the diff of patch set 41 to 42 is shown in the Gerrit Web UI or in the eclipse EGerrit plugin, it is shown correctly (Only the commit message is shown due to the change commit ID).

**Comment by uwolfer** [comment]
This plugin just uses a plain diff between the two commits. So also files affected by the rebase are displayed.

At least the web UI filters out any files not affected by the base change when displaying rebase diffs:
https://bugs.chromium.org/p/gerrit/issues/detail?id=217 (https://gerrit-review.googlesource.com/57086/)

The Eclipse Mylyn plugin has also implemented filtering by its own: https://bugs.eclipse.org/bugs/show_bug.cgi?id=448151 (https://git.eclipse.org/r/39009/).

**Comment by jbaumgar** [comment]
Thanks for your quick response. Would be great if the feature would be implemented. This is currently the only thing that requrires me to use the Web UI instead of the plugin.

**Comment by yehuday** [comment]
is there any update on this issue ?

**Comment by uwolfer** [comment]
@yehuday: No, any updates would be posted in this issue. Feel free to implement it - I'm happy to review pull requests.
